internal/kubernetes/virtualNodeupdater.go

```diff
@@ -89,45 +104,45 @@ func (p *KubernetesProvider) ReconcileNodeFromAdv(event watch.Event) {
 		if err := p.updateFromAdv(*adv); err == nil {
 			klog.Info("node correctly updated from advertisement")
 			break
+		} else {
+			klog.Errorf("node update from advertisement %v failed for reason %v; retry...", adv.Name, err)
 		}
-		klog.Errorf("node update from advertisement %v failed, trying again...", adv.Name)
 	}
 }
 
-// Initialization of the Virtual kubelet, that implies:
-func (p *KubernetesProvider) initVirtualKubelet(adv advtypes.Advertisement) error {
-	klog.Info("vk initializing")
+func (p *KubernetesProvider) ReconcileNodeFromTep(event watch.Event) {
+	tep, ok := event.Object.(*nettypes.TunnelEndpoint)
+	if !ok {
+		klog.Fatal("error in casting tunnelEndpoint")
+	}
+	if event.Type == watch.Deleted {
+		klog.Infof("tunnelEndpoint %v deleted", tep.Name)
```

Comment:
If there is no TEP, should the node be unreacheable?

Reply:
good point, thank you!